aws/table_aws_wafregional_web_acl.go

```diff
@@ -0,0 +1,305 @@
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "arn",
+				Description: "The Amazon Resource Name (ARN) of the entity.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getWafRegionalWebAcl,
+				Transform:   transform.FromField("WebACLArn"),
+			},
+			{
+				Name:        "web_acl_id",
+				Description: "The unique identifier for the Web ACL.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("WebACLId"),
+			},
+			{
+				Name:        "default_action",
+				Description: "The action to perform if none of the Rules contained in the WebACL match.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getWafRegionalWebAcl,
+				Transform:   transform.FromField("DefaultAction.Type"),
+			},
+			{
+				Name:        "metric_name",
+				Description: "A friendly name or description for the metrics for this WebACL.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getWafRegionalWebAcl,
+			},
+			{
+				Name:        "logging_configuration",
+				Description: "The logging configuration for the specified web ACL.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getWafRegionalLoggingConfiguration,
+			},
+			{
+				Name:        "rules",
+				Description: "The Rule statements used to identify the web requests that you want to allow, block, or count.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getWafRegionalWebAcl,
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags associated with the resource.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     listTagsForWafRegionalWebAcl,
+				Transform:   transform.FromField("TagInfoForResource.TagList"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     listTagsForWafRegionalWebAcl,
+				Transform:   transform.FromField("TagInfoForResource.TagList").Transform(wafRegionalWebAclTagListToTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getWafRegionalWebAcl,
+				Transform:   transform.FromField("WebACLArn").Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listWafRegionalWebAcls(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	maxItems := int32(100)
+	params := &wafregional.ListWebACLsInput{}
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxItems {
+			params.Limit = limit
+		}
+	}
```

Comment:
@karanpopat As per the above logic if a user does not provide the limit value in the query, the API will be called with it's default limit value, not the max result value. Because we are not initializing the limit value [in this line](https://github.com/turbot/steampipe-plugin-aws/pull/1660/files#diff-9f6f9a4412b5bbf11c14630b78f88813429b5a4e73c6b3002f64457dd55d3557R117).

Could you please update the logic to get the maximum result per page if a user does not provide any limit value in the query?